src/std/rpc/ws_client/mod.rs

```diff
@@ -155,22 +169,38 @@ fn extrinsic_has_failed(event_details: &EventDetails) -> bool {
 pub struct GetRequestHandler;
 
 impl HandleMessage for GetRequestHandler {
-	fn handle_message(&self, msg: Message, out: Sender, result: ThreadOut<String>) -> WsResult<()> {
+	type ThreadMessage = RpcMessage;
+
+	fn handle_message(
+		&self,
+		msg: Message,
+		out: Sender,
+		result: ThreadOut<Self::ThreadMessage>,
+	) -> WsResult<()> {
 		out.close(CloseCode::Normal)
 			.unwrap_or_else(|_| warn!("Could not close Websocket normally"));
 
 		info!("Got get_request_msg {}", msg);
 		let result_str = serde_json::from_str(msg.as_text()?)
-			.map(|v: serde_json::Value| v["result"].to_string())
-			.map_err(|e| Box::new(RpcClientError::Serde(e)))?;
+			.map(|v: serde_json::Value| Some(v["result"].to_string()))
+			.map_err(RpcClientError::Serde);
 
-		result.send(result_str).map_err(|e| Box::new(RpcClientError::Send(e)).into())
+		result
+			.send(result_str)
+			.map_err(|e| Box::new(RpcClientError::Send(format!("{:?}", e))).into())
 	}
 }
 #[derive(Default, Debug, PartialEq, Eq, Clone)]
 pub struct SubscriptionHandler {}
 impl HandleMessage for SubscriptionHandler {
-	fn handle_message(&self, msg: Message, out: Sender, result: ThreadOut<String>) -> WsResult<()> {
+	type ThreadMessage = String;
```

Comment:
Subscriptions still return a String only because subscription is an ongoing thing and multiple answers are expected. 
Therefore, it should be fine to handle Errors and None values internally and to simply log them.